### PR TITLE
chore: convert controllers to typescript

### LINKED ISSUE #672: Convert top-level files in `src/controllers` to TS
**Is your feature request related to a problem? Please describe.**
Convert top-level JS files in `src/controllers` to TS

**Describe the solution you'd like**
Convert [these top-level files](https://github.com/nkowaokwu/igbo_api/tree/master/src/controllers) (ignoring any files in utils) to TS

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by IfyNdu**
## Describe your changes
- [x] Convert top level JS in `/controllers` to TS
- [x] install relevant type definitions for modules that don't ship their own types
- [x] Create new global and file scoped `types` where necessary
- [x] Relocate and expose global `types` for consumption by existing files for eg `IgboAPIRequest`, `Word` and `Example` types. (If you are happy with this I can go ahead and delete types that are no longer needed as they were locally scoped)
- [x] Minimal but inevitable refactoring in order to make files strongly `typed`

## Issue ticket number and link
Closed #672 

## How Has This Been Tested?
Regression testing

## Screenshots (if appropriate):
Screenshot 2023-07-27 at 00 01 07

### CODE REVIEW

File: src/dictionaries/ig-en/ig-en.json
Comment (ijemmao) [inline src/dictionaries/ig-en/ig-en.json:1]: doesn't seem like this file needs to have all these changes - can you revert the changes in this file?

File: .gitignore [hunk 1]
```diff
@@ -10,6 +10,7 @@ lerna-debug.log*
 dump/
 db/
 mongos/
+dump.rdb
```
Comment (ijemmao) [inline .gitignore:13]: why is this added?

**Review by ijemmao — CHANGES_REQUESTED** [review]
looks really great! just a few comments

File: src/dictionaries/ig-en/ig-en.json
Comment (IfyNdu) [reply src/dictionaries/ig-en/ig-en.json:1]: Oops my bad, forgot to revert changes. I prettified it just to get an idea of the data model. Refactored, Thanks,

File: .gitignore (hunk 1 above)
Comment (IfyNdu) [reply .gitignore]: @ijemmao 
Said file must have been generated whilst setting up the project. Refactored it, Thanks!

File: src/types/example.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  meaning?: string;
+  nsibidi?: string;
+  nsibidiCharacters: string[];
+  associatedWords: string[];
+  associatedDefinitionsSchemas: string[];
+  pronunciations: Pronounciation[];
+  updatedAt: Date;
+};
+
+export type Pronounciation = {
```
Comment (ijemmao) [inline src/types/example.ts:14]:
```suggestion
export type Pronunciation = {
```

also nit: i don't think this type needs to be exported since it's not being used

**Review by ijemmao — CHANGES_REQUESTED** [review]

File: src/types/example.ts (hunk 2 above)
Comment (IfyNdu) [reply src/types/example.ts]: @ijemmao 
Refactored, Thanks!

**Review by ijemmao — DISMISSED** [review]
amazing work, well done!
